### PR TITLE
Moved the structure of the keras documentation into a separate file.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
### Summary

We had a number of issues about the documentation recently. Often related to `autogen.py`. This indicates that some work must be done to simplify this file and making it more accessible. 

### Related Issues

#11382
#11375
#11306
#11673
#11623

### Changes:

- Separating the parsing and the declaration of the structure of the docs. They should be independant.
- Making sure there is no code between functions declarations, this is easy to miss and unexpected. Logic should be put in functions or in `if __name__ == '__main__':`.
- Raise an error when users are not building the docs with the tensorflow backend to avoid them wondering why this is not working. The docstrings are only present in `tensorflow_backend.py`.
- The docs can now be built either by calling `python autogen.py` or by calling `docs.autogen.generate()`.

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -7,6 +7,6 @@ Our documentation uses extended Markdown, as implemented by [MkDocs](http://mkdo
 
 - install MkDocs: `pip install mkdocs`
 - `cd` to the `docs/` folder and run:
-    - `python autogen.py`
```
Comment (Dref360) [inline docs/README.md]: Sometimes, users forget to add Keras home directory to their path.

I suggest : 
```python
pip install mkdocs
export PYTHONPATH=$PYTHONPATH:$PWD
cd docs/
...
```

Comment (gabrieldemarmiesse) [reply docs/README.md]: Done. I prefered the `pip install -e .` to be consistent with the `keras/CONTRIBUTING.md` file.

**Comment by gabrieldemarmiesse** [comment]
This PR is also a first step towards using the same codebase for documentation generation in keras-contrib and keras.

**Comment by gabrieldemarmiesse** [comment]
Thanks @Dref360 !

**Pull request merged**